### ISSUE TITLE
Page just shows "Loading..."

### ISSUE THREAD

**Issue by evdubs** [RangerMauve/load-dat-page#3]
To start, I can see my dat on [datbase](https://datbase.org/dat://8eed8dc41ab7ff1ac84ce4c4cf364bf04218b719b601c802ca6bcad46421b748).

I am looking at the following URL:

https://ranger.mauve.moe/load-dat-page/?url=dat%3A%2F%2F8eed8dc41ab7ff1ac84ce4c4cf364bf04218b719b601c802ca6bcad46421b748

On `load-dat-page`, it just shows "Loading..." without ever showing the dat contents. There is no `index.html` associated with this dat.

On IRC, RangerMauve asked me to paste the browser console. Here's what that shows in Firefox:

```
Error creating socket to local discovery server DOMException: "The operation is insecure." bundle.js:16856:7
    DiscoverySwarmStreamWebsocket https://ranger.mauve.moe/load-dat-page/bundle.js:16856
    DiscoverySwarmWeb https://ranger.mauve.moe/load-dat-page/bundle.js:16756
    Dat https://ranger.mauve.moe/load-dat-page/bundle.js:14516
    [3]< https://ranger.mauve.moe/load-dat-page/bundle.js:255
    o https://ranger.mauve.moe/load-dat-page/bundle.js:1
    o https://ranger.mauve.moe/load-dat-page/bundle.js:1
    [4]< https://ranger.mauve.moe/load-dat-page/bundle.js:491
    o https://ranger.mauve.moe/load-dat-page/bundle.js:1
    r https://ranger.mauve.moe/load-dat-page/bundle.js:1
    <anonymous> https://ranger.mauve.moe/load-dat-page/bundle.js:1
```

I was also asked to paste this URL to hashbase. I did not do that.

**Comment by RangerMauve** [comment]
Hey, thanks for opening the issue!

Thanks to this I found a bug in [resolve-dat-path](https://github.com/RangerMauve/resolve-dat-path).

Your Dat seems to be empty, so I'm not having luck debugging further. 😅

I just released a new version of load-dat-page. Let me know if that changes anything.

**Comment by evdubs** [comment]
I don't know what makes `dat share` unable to establish connections after idling for a while, but it should be available now if you want to check.

**Comment by RangerMauve** [comment]
I can't load your dat in Beaker either. What sort of network are you running `dat share` on?

**Comment by evdubs** [comment]
I am running a pair of `dat share` instances on my home computer.

Edit: this sits behind a conventional router/firewall.

**Comment by RangerMauve** [comment]
Try forwarding port 3283 to your computer through the firewall. That's the port that Dat uses by default

**Comment by evdubs** [comment]
Does it look any better now?

**Comment by evdubs** [comment]
Also, I see listening on 3282 and not 3283.

**Comment by RangerMauve** [comment]
Whoops, I meant to type 3282. 😅

**Comment by evdubs** [comment]
Here's another URL that seems to be able to maintain more peers

https://ranger.mauve.moe/load-dat-page/?url=dat%3A%2F%2F6a797780a33db5b65fe08e5ba5a8dcf0277ba9d508d064efda08458db1e1a3da

**Comment by RangerMauve** [comment]
Oh boy.

Thank you so much for reporting this issue!

At some point I introduced a bug which prevented `load-dat-page` from rendering directories! It seems to all be working for me now.

Lemme know if it's working for you too.

**Comment by evdubs** [comment]
Seems to work for the top level directory but I can't see the contents of a directory. Can you see folder contents? Also on Firefox, I can't see the contents of dat.json.

**Comment by RangerMauve** [comment]
Are there any new errors in the console?

**Comment by RangerMauve** [comment]
I've touched up the navigation a little.

**Comment by RangerMauve** [comment]
Thank you for the patience by the way! This is super experimental so there's a lot of bugs. 😅

**Comment by evdubs** [comment]
I see these errors when loading [this URL](https://ranger.mauve.moe/load-dat-page/?url=dat%3A%2F%2F6a797780a33db5b65fe08e5ba5a8dcf0277ba9d508d064efda08458db1e1a3da) in Chromium:
```
WebSocket connection to 'ws://localhost:3472/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
WebSocketStream @ bundle.js:42437
```

```
Uncaught Error: Unhandled "error" event. ([object Event])
    at Duplexify.emit (bundle.js:11243)
    at Duplexify._destroy (bundle.js:17106)
    at bundle.js:17097
    at Item.run (bundle.js:32467)
    at drainQueue (bundle.js:32437)
```

**Comment by RangerMauve** [comment]
Which URL is this with?

**Comment by evdubs** [comment]
Please see the edited comment.

**Comment by RangerMauve** [comment]
I'm having trouble loading the URL in beaker. I don't think that this is necessarily a load-dat-page, I think this might just be trouble with getting peers to connect to your dat. :/

Not 100% sure what the best path forward would be. Maybe ask on IRC for what to do about connectivity issues?

**Comment by evdubs** [comment]
Does it seem strange to you that the following URLs show up on datbase as having "sources available" but Beaker can't view them? 

https://datbase.org/view?query=dat://6a797780a33db5b65fe08e5ba5a8dcf0277ba9d508d064efda08458db1e1a3da
https://datbase.org/view?query=dat://8eed8dc41ab7ff1ac84ce4c4cf364bf04218b719b601c802ca6bcad46421b748

I was able to see the contents of `dat://6a797780a33db5b65fe08e5ba5a8dcf0277ba9d508d064efda08458db1e1a3da` on Beaker for a moment, but now it doesn't work.

**Comment by evdubs** [comment]
Also, I am still able to see the following repositories on http://ranger.mauve.moe/dat-js-example

dat://6a797780a33db5b65fe08e5ba5a8dcf0277ba9d508d064efda08458db1e1a3da
dat://8eed8dc41ab7ff1ac84ce4c4cf364bf04218b719b601c802ca6bcad46421b748

So this is confusing to me that some things work, others don't, etc.

**Comment by RangerMauve** [comment]
That's super weird. I'm honestly not sure why that would be the case. 🙃

**Comment by evdubs** [comment]
So one "last" thing with this issue is that I can't seem to open the CSVs in the directories. Is it expected that `load-dat-page` can serve CSVs? Right now, they just seem to get stuck at "Loading..."

**Comment by RangerMauve** [comment]
I think it should be rendering the CSV files as text.

https://github.com/RangerMauve/load-dat-page/blob/master/contentLoader.js#L159

How large are they?

**Comment by evdubs** [comment]
They're between 700kB and 2.5MB.

**Comment by RangerMauve** [comment]
The smaller ones should be able to load up pretty quickly

**Comment by evdubs** [comment]
Trying to open http://ranger.mauve.moe/load-dat-page/?url=dat%3A%2F%2F6a797780a33db5b65fe08e5ba5a8dcf0277ba9d508d064efda08458db1e1a3da%2Feps-estimate%2F2017-10-26.csv (a 6.4kB file) just seems to get stuck at "Loading..."

**Comment by RangerMauve** [comment]
Yeah, not sure what's up with that. I think I'll need to do more debugging to see why.

**Comment by evdubs** [comment]
Hi,
I am able to see the contents of:

https://ranger.mauve.moe/load-dat-page/?url=dat%3A%2F%2F8eed8dc41ab7ff1ac84ce4c4cf364bf04218b719b601c802ca6bcad46421b748

but I cannot see the contents of (it just shows "Loading..."):

https://ranger.mauve.moe/load-dat-page/?url=dat%3A%2F%2F8eed8dc41ab7ff1ac84ce4c4cf364bf04218b719b601c802ca6bcad46421b748%2Foption-chain

Things seem to have regressed?

**Comment by RangerMauve** [comment]
That's really strange. I'm not sure why it'd be doing that. :/

**Comment by RangerMauve** [comment]
I'm not seeing any available peers

**Comment by RangerMauve** [comment]
Wait, NVM. It's just full of CSV files. No clue why that wouldn't render, then. :S

**Comment by RangerMauve** [comment]
Oh wait! It worked. I just had to wait for a while and the CSV files are showing up.

![image](https://user-images.githubusercontent.com/911495/58330707-7edf2400-7e05-11e9-90f7-2274f2e4fda8.png)

**Comment by evdubs** [comment]
How long was "a while"? I have let it sit for hours and it still shows "Loading..." The console doesn't seem to show anything useful.

**Comment by RangerMauve** [comment]
Weird. It loaded for me after I opened up Beaker on my machine. What do you have seeding those files?

**Comment by evdubs** [comment]
I am using Dat Desktop to seed these files. I recently tried to download a CSV and it works after a bit of waiting on "Loading..." so that's cool! I guess the waiting is just the process of fetching the data remotely and sending to the client?

Edit: I think there may be some weirdness with my port forwarding and using Dat Desktop to seed two dats at the same time.

**Comment by RangerMauve** [comment]
I think Dat desktop is being deprecated because it's stability isn't too great at the moment. How's the stability if you use Beaker, the CLI, or [dat-store](https://github.com/datproject/dat-store)?

**Comment by evdubs** [comment]
I tried sharing my dats from Beaker and there doesn't seem to be an easy way to share an existing dat? Maybe I am doing something wrong? I tried "Create New" and "Share" and neither of those options seem to pick up an existing dat to share.

My previous attempts at using `dat share` did not seem as stable as using Dat Desktop. Maybe I'll give dat-store a look, but I prefer having a GUI client.

**Comment by evdubs** [comment]
Also, I seem to be unable to load my dats in Beaker even though I can see them on datbase and load-dat-page.

Edit: this problem persists even if I use `dat share` for the dats. Shall I open an issue for this on beaker?

**Comment by evdubs** [comment]
Regarding the recent comments, I added a comment to https://github.com/beakerbrowser/beaker/issues/1424
